### DISCUSSION TITLE
Duplicated database queries during filtering

### DISCUSSION THREAD

**Discussion by lukogites** [rappasoft/laravel-livewire-tables#565] [Q&A] [answered]
I wrote filter:

`'leave_type_id' => Filter::make('Typ urlopu/nieobecności')
                ->select(LeaveType::select(['id', 'name'])->pluck('name', 'id')->toArray()),`

I go to the datatable page, open Laravel debugbar and see 6 database queries including 5 duplicates. 

![Screenshot](https://user-images.githubusercontent.com/47400021/142493929-f7a67e0c-5970-4852-bf54-d186c812bca3.png)

Why does this happen and how can it be prevented?

**Comment by rizkhal** [comment, marked as answer]
That's because you create query `LeaveType::select(['id', 'name'])->pluck('name', 'id')->toArray())` inside the filters method..

See `WithFilters.php` file [here](https://github.com/rappasoft/laravel-livewire-tables/blob/5ec4a23cc7c937d305afb64fc32a57a5881f824e/src/Traits/WithFilters.php#L117).

To solve this issue, you need to create `query` outside the `filters` method and place the result of your `query`

For example
```php
public function leaveTypes(): array
{
   return LeaveType::select(['id', 'name'])->pluck('name', 'id')->toArray();
}

public function filters(): array
{
   return [
       'leave' => Filter::make('Leave')->select($this->leaveTypes()),
   ];
}
```

**Reply by lukogites** [reply]
I did as you recommended and the effect is the same. 5 duplicate database queries. :(

**Reply by rizkhal** [reply]
oh sory my mistake, you need to use property instead of method and place it on mout method.

looks like
```php
public function mount()
{
  $this->options = LeaveType::select(['id', 'name'])->pluck('name', 'id')->toArray();
}
```

then place options property on your filter select

**Reply by lukogites** [reply]
Now it works perfectly. Thank you so much for your help!
